### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by MacioJanisz**

### CODE REVIEW

**Review by Rommelua — CHANGES_REQUESTED** [review]
Java Cl checks failed. Before making a Pull Request (or a new commit to an already existing PR), you need to write the mvn clean package command in the terminal and make sure that all checks pass. After that, the mentor will be able to review the homework

image

**Comment by MacioJanisz** [comment]
I fix mvn error and add interface

File: src/main/java/core/basesyntax/Circle.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    public Circle(double radius, String color) {
+        super(color);
+        this.radius = radius;
+
+    }
+
+    public double getRadius() {
+
+        return radius;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Circle.java:14]: image

File: src/main/java/core/basesyntax/Circle.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    public void setRadius(double radius) {
+        this.radius = radius;
+    }
+
+    @Override
+    public double getArea() {
+        return Math.PI * Math.pow(radius, 2);
+
+    }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Circle.java:25]: Remove all redundant empty lines, be careful :)

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private static final String[] COLORS = {"red", "green", "blue", "yellow", "purple"};
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: Let's create **enum Color** in a separate file

File: src/main/java/core/basesyntax/Figure.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public abstract double getArea();
+
+    public abstract void draw();
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Figure.java:16]:
```suggestion
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                    colorSupplier.getRandomColor());
+            case 4 ->
+                    new IsoscelesTrapezoid(random.nextDouble(), random.nextDouble(),
+                            random.nextDouble(), colorSupplier.getRandomColor());
+            default -> getDefaultFigure();
+        };
+    }
+
+    public Figure getDefaultFigure() {
+        int radiusCircle = 10;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/FigureSupplier.java:27]: image

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        return new Circle(radiusCircle, "white");
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/FigureSupplier.java:28]:
```suggestion
        return new Circle(DEFAULT_RADIUS, Color.WHITE);
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Comment by MacioJanisz** [comment]
I remove empty lines, change circle color declatrions and add enum colors.

File: src/main/java/core/basesyntax/Figure.java (hunk 4 above)
Comment (Rommelua) [reply src/main/java/core/basesyntax/Figure.java:16]: delete these 2 abstract mrthods - no any reason to have them in this class

File: src/main/java/core/basesyntax/FigureSupplier.java (hunk 5 above)
Comment (Rommelua) [reply src/main/java/core/basesyntax/FigureSupplier.java:28]: you can delete `int radiusCircle = 10;`

**Review by Rommelua — APPROVED** [review]
Almost good job, approved

**Comment by MacioJanisz** [comment]
Thank You, I change this two things

**Pull request closed without merging**